cachito/workers/tasks/rubygems.py

```diff
@@ -16,3 +23,61 @@ def cleanup_rubygems_request(request_id):
         "username": get_rubygems_nexus_username(request_id),
     }
     nexus.execute_script("rubygems_cleanup", payload)
+
+
+def _get_config_file_for_given_package(
+    dependencies, bundle_dir, package_source_dir, rubygems_hosted_url
+):
+    """
+    Get Bundler config file.
+
+    Returns a Bundler config file with a mirror set for RubyGems dependencies pointing to
+    `rubygems_hosted_repo` URL. All GIT dependencies are configured to be replaced by local git
+     repos.
+
+    :param dependencies: an array of dependencies (dictionaries) with keys
+        "name": package name,
+        "path": an absolute path to a locally downloaded git repo,
+        "kind": dependency kind
+    :param bundle_dir: and absolute path to a bundle root directory
+    :param package_source_dir: the root directory of given package
+    :param rubygems_hosted_url: URL pointing to a request specific RubyGems hosted repo with
+     hardcoded user credentials
+    :return: dict with "content", "path" and "type" keys
+    """
+    base_config = dedent(
+        f"""
+        # Make sure gems aren’t installed system-wide but only for the given project
+        BUNDLE_PATH: "vendor/bundle"
+        # Sets mirror for all RubyGems sources
+        BUNDLE_MIRROR__ALL: "{rubygems_hosted_url}"
+        # Turn off the probing
+        BUNDLE_MIRROR__ALL__FALLBACK_TIMEOUT: "false"
+        # Install only ruby platform gems (=> gems with native extensions are compiled from source).
+        # All gems should be platform independent already, so why not keep it here.
+        BUNDLE_FORCE_RUBY_PLATFORM: "true"
+        BUNDLE_DEPLOYMENT: "true"
+        # Defaults to true when deployment is set to true
+        BUNDLE_FROZEN: "true"
+    """
+    )
+
+    config = [base_config]
+    for dependency in dependencies:
+        if dependency["kind"] == "GIT":
+            # These substitutions are required by Bundler
+            name = dependency["name"].upper().replace("-", "___")
```

Comment:
Is this supposed to be 3 underscores? Just asking because other things seem to use 2 underscores

Reply:
Yes, from manual testing I can say it matches substitutions that [Bundler does for credentials](https://bundler.io/man/bundle-config.1.html#CREDENTIALS-FOR-GEM-SOURCES), where `-` is mapped to `___` (3 underscores) and `.` to `__` (2 underscores - I forgot about it)